Blog post SCM/CI integration roll-out announcement (#374)
openSUSE/obs-landing: _posts/development/2022-09-21-trigger-workflow-rollout.md

---
layout: post

> title: SCM/CI Integration in OBS, a Definitive Feature

Reviewer: ```suggestion
title: General availability of the SCM/CI Integration feature
```

category: development
---

> On May 2021 we took our first steps towards the continuous integration between
> OBS and different SCMs. Thanks to the valuable feedback we received from users

Reviewer: ```suggestion
In May of 2021 we took our first steps towards putting OBS builds into the continuous integration cycle with the SCM/CI feature. Thanks to the valuable feedback we received from users
```

who started as early testers and became heavy users of the feature, we are

> ready to roll it out for every single user in OBS.

Reviewer: ```suggestion
ready today to push it out of the beta program, making it generally available in OBS.
```

---
layout: post
title: General availability of the SCM/CI Integration feature
category: development
---

In May of 2021 we took our first steps towards putting OBS builds into the
continuous integration cycle with the SCM/CI feature. Thanks to the valuable
feedback we received from users who started as early testers and became heavy
users of the feature, we are ready today to push it out of the beta program,
making it generally available in OBS.

### What Is This About?

> With the feature we are pushing out, you can take advantage of source code
> management (SCM) systems like GitHub or GitLab to manage your packages sources
> and then, you can integrate those sources with OBS to run different workflows.

Reviewer: ```suggestion
With the SCM/CI feature you can take advantage of source code management (SCM) systems like GitHub or GitLab to manage your package sources and integrate OBS to run different continuous integration workflows.
```

You can set up workflows to
build a package in OBS after each pull request and then report back the result to the SCM,
to rebuild a package after each change (push event) in a branch,
or to create a package in OBS for every software release using Git tags,
among others.

### Haven't You Started With The Integration Yet?

All you need to know is in [our
documentation](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html).
Give it a try!

> 

Reviewer: I would add another screenshot of the github side. Like

![Screenshot from 2022-09-20 16-23-16](https://user-images.githubusercontent.com/514785/191283842-d14fe6e3-f7c8-4fa7-9268-39d9ce886a40.png)

Author: Done, but I moved the images to _What Is This About?_ section. They make more sense there where we mention workflows and SCM reports.

<figure>
  <img src="/images/posts/sprint_126_workflow_runs.png" alt="Workflow runs" />
  <figcaption>Workflow runs</figcaption>
</figure>

You can learn more about our journey by reading previous blog posts.

> {% include partials/_series-of-posts-about-scm-integration.md %}

Reviewer: I wouldn't add this. It doesn't matter how we got here, it matters that we are here.

Author: I don't include the partial on this post then.

users of the feature, we are ready today to push it out of the beta program,
making it generally available in OBS.

### What Is This About?

With the SCM/CI feature you can take advantage of source code management (SCM)
systems like GitHub or GitLab to manage your package sources and integrate OBS
to run different continuous integration workflows.

You can set up workflows to
build a package in OBS after each pull request and then report back the result to the SCM,
to rebuild a package after each change (push event) in a branch,
or to create a package in OBS for every software release using Git tags,
among others.

<figure>
  <img src="/images/posts/sprint_126_workflow_runs.png" alt="Workflow runs in OBS" />
  <figcaption>Details about the workflow runs in OBS</figcaption>
</figure>

<figure>
  <img src="/images/posts/sprint_126_checks.png" alt="Checks results on GitHub" />
  <figcaption>Checks results on GitHub</figcaption>
</figure>

### Haven't You Started With The Integration Yet?

All you need to know is in [our
documentation](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html).
Give it a try!

> You can learn more about our journey by reading our previous blog posts about the SCM/CI integration feature.

Author: No, I decided to leave it there to invite people to read more... But let me remove it and organize the paragraphs differently.

> You can learn more about our journey by reading our previous blog posts about the SCM/CI integration feature.

Author: Ok, I removed that sentence and merge the section with the previous one. Let me know what you think, please.
